dvc/repo/__init__.py

```diff
@@ -426,10 +435,11 @@ def find_outs_by_path(self, path, outs=None, recursive=False):
             outs = [out for stage in self.stages for out in stage.outs]
 
         abs_path = os.path.abspath(path)
+        path_info = PathInfo(abs_path)
         is_dir = self.tree.isdir(abs_path)
 
         def func(out):
-            if out.scheme == "local" and out.fspath == abs_path:
+            if out.scheme == "local" and path_info.isin_or_eq(out.path_info):
                 return True
 
             if is_dir and recursive and out.path_info.isin(abs_path):
```

Comment:
```suggestion
            if is_dir and recursive and out.path_info.isin(path_info):
```
`.isin()` will make a `PathInfo` anyway.